compiler/crates/js-config-loader/tests/test.rs

```diff
@@ -0,0 +1,184 @@
+use js_config_loader::{search, ConfigError, ErrorCode};
+use serde::Deserialize;
+use std::fs::create_dir_all;
+use tempfile::tempdir;
+
+#[derive(Debug, Deserialize)]
+struct TestConfig {
+    name: String,
+}
+
+#[test]
+fn package_json_and_rc_same_dir() {
+    let dir = tempdir().unwrap();
+    let dir_d = dir.path().join("a/b/c/d");
+    let dir_f = dir.path().join("a/b/c/d/e/f");
+    create_dir_all(&dir_f).unwrap();
+
+    std::fs::write(dir_d.join("package.json"), r#"{ "false": "hope" }"#).unwrap();
+    std::fs::write(dir_d.join(".foorc"), r#"{ "name": "correct" }"#).unwrap();
+
+    let config = search::<TestConfig>("foo", &dir_f).unwrap().unwrap();
+    assert_eq!(config.value.name, "correct");
+}
+
+#[test]
+fn rc_json_invalid_syntax() {
+    let dir = tempdir().unwrap();
+    let dir_d = dir.path().join("a/b/c/d");
+    let dir_f = dir.path().join("a/b/c/d/e/f");
+    create_dir_all(&dir_f).unwrap();
+
+    std::fs::write(dir_d.join(".foorc"), "invalid").unwrap();
+
+    match search::<TestConfig>("foo", &dir_f) {
```

Comment:
This is the API, this crate is independent of the shape of the config, but returns a typed result. It just needs the value to implement `Deserialize`.

I really love `serde` <3